analyzer/codechecker_analyzer/cmd/parse.py

```diff
@@ -410,55 +425,81 @@ def get_output_file_path(default_file_name: str) -> Optional[str]:
             file_paths = specifed_file_paths or file_paths
 
         for file_path in file_paths:
-            reports = report_file.get_reports(
-                file_path, context.checker_labels, file_cache)
-
-            reports = reports_helper.skip(
-                reports, processed_path_hashes, skip_handlers, suppr_handler,
-                src_comment_status_filter)
-
-            statistics.num_of_analyzer_result_files += 1
-            for report in reports:
-                if report.changed_files:
-                    changed_files.update(report.changed_files)
-
-                statistics.add_report(report)
-
-                if trim_path_prefixes:
-                    report.trim_path_prefixes(trim_path_prefixes)
-
-            all_reports.extend(reports)
-
-            # Print reports continously.
-            if not export:
-                file_report_map = plaintext.get_file_report_map(
-                    reports, file_path, metadata)
-                plaintext.convert(
-                    file_report_map, processed_file_paths, print_steps)
-            elif export == 'html':
-                print(f"Parsing input file '{file_path}'.")
-                report_to_html.convert(
-                    file_path, reports, output_dir_path,
-                    html_builder)
-
-    if export is None:  # Plain text output
+            if 'summary' in args:
```

Comment:
I'm not to hot about endlessly elongating functions like this, but I've seen arguments for it too, so, if you really insist, sure, but I'd prefer to have a dedicated function for this to lessen the scope of it. That would make unit testing easier as well.

On another note, #3843 does much of the same thing as this patch, can we not merge those implementations?